protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MagnetTools/index.tsx

```diff
@@ -1,3 +1,95 @@
-export function MagnetTools(): JSX.Element {
-  return <div>TODO: wire this up</div>
+import { useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import {
+  Box,
+  COLORS,
+  DIRECTION_COLUMN,
+  Flex,
+  ListItem,
+  SPACING,
+  StyledText,
+} from '@opentrons/components'
+import { MAGNETIC_MODULE_V1 } from '@opentrons/shared-data'
+import {
+  MAX_ENGAGE_HEIGHT_V1,
+  MAX_ENGAGE_HEIGHT_V2,
+  MIN_ENGAGE_HEIGHT_V1,
+  MIN_ENGAGE_HEIGHT_V2,
+} from '../../../../../../constants'
+import {
+  getMagnetLabwareEngageHeight,
+  getMagneticLabwareOptions,
+} from '../../../../../../ui/modules/selectors'
+import { ToggleExpandStepFormField } from '../../../../../../molecules'
+import { getModuleEntities } from '../../../../../../step-forms/selectors'
+
+import type { StepFormProps } from '../../types'
+
+export function MagnetTools(props: StepFormProps): JSX.Element {
+  const { propsForFields, formData } = props
+  const { t } = useTranslation(['application', 'form', 'protocol_steps'])
+  const moduleLabwareOptions = useSelector(getMagneticLabwareOptions)
+  const moduleEntities = useSelector(getModuleEntities)
+  const defaultEngageHeight = useSelector(getMagnetLabwareEngageHeight)
+  const moduleModel = moduleEntities[formData.moduleId].model
+
+  const mmUnits = t('units.millimeter')
+  const isGen1 = moduleModel === MAGNETIC_MODULE_V1
+  const engageHeightMinMax = isGen1
+    ? t('magnet_height_caption', {
+        low: MIN_ENGAGE_HEIGHT_V1,
+        high: MAX_ENGAGE_HEIGHT_V1,
+      })
+    : t('magnet_height_caption', {
+        low: `${MIN_ENGAGE_HEIGHT_V2} ${mmUnits}`,
+        high: `${MAX_ENGAGE_HEIGHT_V2} ${mmUnits}`,
+      })
+  const engageHeightDefault =
+    defaultEngageHeight != null
+      ? isGen1
+        ? t('magnet_recommended', { default: defaultEngageHeight })
+        : t('magnet_recommended', {
+            default: `${defaultEngageHeight} ${mmUnits}`,
+          })
+      : ''
+  const engageHeightCaption = `${engageHeightMinMax} ${engageHeightDefault}`
+
+  return (
+    <Flex flexDirection={DIRECTION_COLUMN}>
+      <Flex
+        flexDirection={DIRECTION_COLUMN}
+        padding={`${SPACING.spacing16} ${SPACING.spacing16} ${SPACING.spacing12} ${SPACING.spacing16}`}
```

Comment:
nit
`top | left and right | bottom`

```suggestion
        padding={`${SPACING.spacing16} ${SPACING.spacing16} ${SPACING.spacing12}`}
```